### PR TITLE
More typing on context

### PR DESCRIPTION
**Pull request by mccalluc**
I had started this a while ago and then forgot about it. Just some more type annotations.

### CODE REVIEW

**Review by Shoeboxam — APPROVED** [review]

**Comment by mccalluc** [comment]
Seemed like it should be easy, but there is a mypy error. Will investigate:
```
src/opendp/context.py:273: error: Incompatible return value type (got "tuple[Measure, float | None]", expected "tuple[Measure, float | tuple[float, float]]")  [return-value]
src/opendp/context.py:277: error: Incompatible return value type (got "tuple[Measure, tuple[float | None, float]]", expected "tuple[Measure, float | tuple[float, float]]")  [return-value]
Found 2 errors in 1 file (checked 75 source files)
```

File: python/src/opendp/context.py [hunk 1]
```diff
@@ -243,11 +247,6 @@ def loss_of(epsilon=None, delta=None, rho=None, U=None) -> Tuple[Measure, Union[
     :param U: The type of the privacy parameter; Inferred if not provided.
 
     """
-    if epsilon is None and rho is None:
-        raise ValueError("Either epsilon or rho must be specified.")
-    if epsilon is None and delta is not None:
-        raise ValueError("Epsilon must be specified if delta is given.")
-
```
Comment (Shoeboxam) [inline python/src/opendp/context.py:250]: Why remove these checks? It is valid for `rho` to be the only argument.

File: python/src/opendp/context.py [hunk 2]
```diff
@@ -262,6 +261,9 @@ def range_warning(name, value, info_level, warn_level):
         U = RuntimeType.parse_or_infer(U, rho)
         return zero_concentrated_divergence(T=U), rho
 
+    if epsilon is None:
+        raise ValueError("Either epsilon or rho must be specified.")
```
Comment (Shoeboxam) [inline python/src/opendp/context.py:265]: Won't this raise if I call it like `loss_of(rho=.5)`?

**Review by Shoeboxam — DISMISSED** [review]

File: python/src/opendp/context.py (hunk 2 above)
Comment (mccalluc) [reply python/src/opendp/context.py:265]: Unless I'm confused, no: If `rho` is set we'll return zero_concentrated_divergence just above.

File: python/src/opendp/context.py (hunk 1 above)
Comment (mccalluc) [reply python/src/opendp/context.py:250]: The prompt for doing this is that the annotation additions in the earlier commit in the PR were causing mypy failures later in this function. The checks here are a little too complicated, and mypy was thinking that `None`s could come through. After replacing these with just one guard below, mypy is happy.

**Review by raprasad — APPROVED** [review]
Approved.

**Review by raprasad — APPROVED** [review]
approved

**Pull request merged**